cmd/kubeadm/app/features/features.go

```diff
@@ -32,12 +32,15 @@ const (
 	IPv6DualStack = "IPv6DualStack"
 	// PublicKeysECDSA is expected to be alpha in v1.19
 	PublicKeysECDSA = "PublicKeysECDSA"
+	// InPlacePodVerticalScaling is expected to be alpha in v1.19
+	InPlacePodVerticalScaling = "InPlacePodVerticalScaling"
 )
 
 // InitFeatureGates are the default feature gates for the init command
 var InitFeatureGates = FeatureList{
-	IPv6DualStack:   {FeatureSpec: featuregate.FeatureSpec{Default: false, PreRelease: featuregate.Alpha}},
-	PublicKeysECDSA: {FeatureSpec: featuregate.FeatureSpec{Default: false, PreRelease: featuregate.Alpha}},
+	IPv6DualStack:             {FeatureSpec: featuregate.FeatureSpec{Default: false, PreRelease: featuregate.Alpha}},
+	PublicKeysECDSA:           {FeatureSpec: featuregate.FeatureSpec{Default: false, PreRelease: featuregate.Alpha}},
+	InPlacePodVerticalScaling: {FeatureSpec: featuregate.FeatureSpec{Default: false, PreRelease: featuregate.Alpha}},
```

Comment:
you can omit the cmd/kubeadm changes from this PR.
InitFeatureGates contains mostly kubeadm specific features (and not k8s features). it is still possible for users to configure `--feature-gates=InPlacePodVerticalScaling`, for e.g. the kubelet.

Reply:
Looking back into history, gates such as Auditing, DynamicKubelet, CoreDNS have lived here. Weren't those k8s features being introduced?

Reply:
yes, but those features also required changes in the kubeadm logic, similarly to the existing `IPv6DualStack`.

`InPlacePodVerticalScaling` is not required to be present to this list as it is a scalability feature that users can pipe trough kubeadm configuration to the components that need it, such as the kubelet.

Reply:
thanks for the explanation. i've backed out that change in commit https://github.com/kubernetes/kubernetes/pull/92127/commits/113cf1c2ccfb154fd5a5e49de2a8d5f8bd3c4c6f

Reply:
thank you.

/remove-area kubeadm
/remove-sig cluster-lifecycle